src/datahike/datom.cljc

```diff
@@ -101,7 +101,7 @@
       (combine-hashes (hash (.-v d)))))
 
 (defn- equiv-datom [^Datom d ^Datom o]
-  (and (== (.-e d) (.-e o))
+  (and (== (.-e d) (.-e o)) ;; why ==?
```

Comment:
We inherited this from DataScript.

Reply:
Apparently it's the type independent version of `=`. Seems counterintuitive to me though that `==` is a weaker equality than `=`.
I'd say `==` makes sense here though. I will remove the comment.

Reply:
I am not sure why `==` is needed here. This only makes sense if the entity-id provided could sometimes be a floating point type on the JVM. In JS there is only one number type.

Reply:
Do you think, we should should just use `=` then?
When I think about it, it would make sense to me since in other places - e.g. the `cmp-datoms-eavt` function etc. - the compiler fails if one of the entity ids is not of type `Integer`.

Reply:
Addressed in issue #385